pipeline-runner/steps/deploy.go

```diff
@@ -99,27 +101,36 @@ func (cli *DeployStepImplementation) deployToEnv(appName, env string, pipelineIn
 		pipelineInfo.PipelineArguments.JobName,
 		pipelineInfo.PipelineArguments.ImageTag,
 		pipelineInfo.PipelineArguments.Branch,
-		pipelineInfo.DeployEnvironmentComponentImages[env],
-		env,
+		pipelineInfo.DeployEnvironmentComponentImages[envName],
+		envName,
 		defaultEnvVars,
 		radixApplicationHash,
 		buildSecretHash,
 		pipelineInfo.PrepareBuildContext,
 		pipelineInfo.PipelineArguments.ComponentsToDeploy)
 
 	if err != nil {
-		return fmt.Errorf("failed to create radix deployments objects for app %s. %v", appName, err)
+		return fmt.Errorf("failed to create Radix deployments for app %s. %w", appName, err)
 	}
 
-	err = cli.namespaceWatcher.WaitFor(utils.GetEnvironmentNamespace(cli.GetAppName(), env))
-	if err != nil {
-		return fmt.Errorf("failed to get environment namespace, %s, for app %s. %v", env, appName, err)
+	namespace := utils.GetEnvironmentNamespace(cli.GetAppName(), envName)
+	if err = cli.namespaceWatcher.WaitFor(namespace); err != nil {
+		return fmt.Errorf("failed to get environment namespace %s, for app %s. %w", namespace, appName, err)
 	}
 
-	log.Info().Msgf("Apply radix deployment %s on env %s", radixDeployment.GetName(), radixDeployment.GetNamespace())
-	_, err = cli.GetRadixclient().RadixV1().RadixDeployments(radixDeployment.GetNamespace()).Create(context.TODO(), radixDeployment, metav1.CreateOptions{})
-	if err != nil {
-		return fmt.Errorf("failed to apply radix deployment for app %s to environment %s. %v", appName, env, err)
+	radixDeploymentName := radixDeployment.GetName()
+	log.Info().Msgf("Apply radix deployment %s on environment %s", radixDeploymentName, envName)
+	if _, err = cli.GetRadixclient().RadixV1().RadixDeployments(radixDeployment.GetNamespace()).Create(context.Background(), radixDeployment, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("failed to apply Radix deployment for app %s to environment %s. %w", appName, envName, err)
+	}
+
+	if err = cli.radixDeploymentWatcher.WaitForActive(namespace, radixDeploymentName); err != nil {
```

Comment:
```suggestion
	if err := cli.radixDeploymentWatcher.WaitForActive(namespace, radixDeploymentName); err != nil {
```
Make the err local to the `if` block